src/middleware/error.middleware.js

```diff
@@ -0,0 +1,16 @@
+import { ApiError } from '../exceptions/api.error.js';
+
+export const errorMiddleware = (error, req, res, next) => {
+  if (error instanceof ApiError) {
+    res.status(error.status).send({
+      message: error.message,
+      errors: error.errors,
+    });
+  }
+
+  res.statusCode = 500;
+
+  res.send({
+    message: 'Server error',
+  });
```

Comment:
The `res.send` call should also be inside an `else` block to ensure it only executes when the error is not an instance of `ApiError`. This prevents sending multiple responses for the same request.